can/io/asc.py

```diff
@@ -197,8 +198,12 @@ def on_message_received(self, msg):
         if msg.is_extended_id:
             arb_id += 'x'
 
-        # Many interfaces start channel numbering at 0 which is invalid
-        channel = msg.channel + 1 if isinstance(msg.channel, int) else self.channel
+        channel = channel2int(msg.channel)
```

Comment:
Should this always be turned into an int? Because SocketCAN's `vcan0`/`can2` are hard to transform into a int. Same applies for BLF as well. Or is that required by the file format? Because then, it should be noted in the code and/or the class docs, because users might expect that Message 1 -> Writer -> Reader -> Message 2 results in Message 1 == Message 2. That should be added somewhere as a warning/note or whatever.

Reply:
Unfortunately the ASC format requires integers (same as BLF). So the channel read back may be different than the one put in. 😞 I still think it's better than not being able to tell the channels apart at all.

Reply:
Yep, you are right. Who designs such short-sighted protocols ... ? Well anyways, we should note that somewhere in the docs at least.